idb/postgres/internal/writer/writer.go

```diff
@@ -209,113 +208,114 @@ func writeAccount(round sdk.Round, address basics.Address, accountData ledgercor
 	} else {
 		// Update account.
 		accountDataJSON :=
-			encoding.EncodeTrimmedLcAccountData(encoding.TrimLcAccountData(accountData))
+			encoding.EncodeTrimmedSdkAccountData(encoding.TrimAccountData(accountData))
 
 		if sigtypeDelta.present {
 			batch.Queue(
 				upsertAccountWithKeytypeStmtName,
-				address[:], accountData.MicroAlgos.Raw, accountData.RewardsBase,
-				accountData.RewardedMicroAlgos.Raw, uint64(round),
+				address[:], accountData.Amount, accountData.RewardBase,
+				accountData.Rewards, uint64(round),
 				sigtypeFunc(sigtypeDelta.value), accountDataJSON)
 		} else {
 			batch.Queue(
 				upsertAccountStmtName,
-				address[:], accountData.MicroAlgos.Raw, accountData.RewardsBase,
-				accountData.RewardedMicroAlgos.Raw, uint64(round),
+				address[:], accountData.Amount, accountData.RewardBase,
+				accountData.Rewards, uint64(round),
 				accountDataJSON)
 		}
 	}
 }
 
-func writeAssetResource(round sdk.Round, resource *ledgercore.AssetResourceRecord, batch *pgx.Batch) {
-	if resource.Params.Deleted {
-		batch.Queue(deleteAssetStmtName, resource.Aidx, resource.Addr[:], round)
+func writeAssetResource(round sdk.Round, resource *models.AssetResourceRecord, batch *pgx.Batch) {
+	addr, _ := sdk.DecodeAddress(resource.Address)
+	if resource.AssetDeleted {
+		batch.Queue(deleteAssetStmtName, resource.AssetIndex, addr[:], round)
 	} else {
-		if resource.Params.Params != nil {
+		if !reflect.DeepEqual(resource.AssetParams, models.AssetParams{}) {
```

Comment:
I haven't researched this usage extensively but I recommend avoiding `reflect.DeepEqual` in the production path whenever possible, because of possible performance implications.